src/hooks/useSelector.js

```diff
@@ -16,33 +16,12 @@ const useIsomorphicLayoutEffect =
 
 const refEquality = (a, b) => a === b
 
-/**
- * A hook to access the redux store's state. This hook takes a selector function
- * as an argument. The selector is called with the store state.
- *
- * This hook takes an optional equality comparison function as the second parameter
- * that allows you to customize the way the selected state is compared to determine
- * whether the component needs to be re-rendered.
- *
- * @param {Function} selector the selector function
- * @param {Function=} equalityFn the function that will be used to determine equality
- *
- * @returns {any} the selected state
- *
- * @example
- *
- * import React from 'react'
- * import { useSelector } from 'react-redux'
- *
- * export const CounterComponent = () => {
- *   const counter = useSelector(state => state.counter)
- *   return <div>{counter}</div>
- * }
- */
-export function useSelector(selector, equalityFn = refEquality) {
-  invariant(selector, `You must pass a selector to useSelectors`)
-
-  const { store, subscription: contextSub } = useReduxContext()
+function useSelectorWithStoreAndSubscription(
```

Comment:
I factored this out to make the diff easier to read, but can inline it again if you'd prefer.